src/file_handler.rs

```diff
@@ -65,24 +65,48 @@ pub mod file_handler {
             Ok(_) => Ok(()),
         }
     }
+}
 
-    pub fn config_location() -> String {
-        match ::env::home_dir() {
-            Some(location) => format!("{}/{}", location.display(), ".eureka"),
-            None => panic!("Could not resolve your $HOME directory"),
-        }
+impl FileManagement for FileHandler {
+    fn file_exists(&self, path: &str) -> bool {
+        fs::metadata(path).is_ok()
     }
 
-    pub fn rm_file(path: String) -> io::Result<()> {
-        if path_exists(&path) {
-            fs::remove_file(&path)?;
+    fn file_rm(&self, file: ConfigFile) -> io::Result<()> {
+        let config_file_path = config_path(file);
+        if self.file_exists(&config_file_path) {
+            fs::remove_file(&config_file_path)?;
             Ok(())
         } else {
             let invalid_path = io::Error::new(
                 ErrorKind::NotFound,
-                format!("Path does not exist: {}", path),
+                format!("Path does not exist: {}", config_file_path),
             );
             Err(invalid_path)
         }
     }
 }
+
+fn config_path(file: ConfigFile) -> String {
+    let file_name = match file {
+        ConfigFile::Repo => CONFIG_REPO.to_string(),
+        ConfigFile::Editor => CONFIG_EDITOR.to_string(),
+    };
+
+    match env::home_dir() {
+        Some(location) => format!(
+            "{home}/{eureka}/{file_name}",
+            home = location.display(),
+            eureka = ".eureka",
+            file_name = file_name
+        ),
+        None => panic!("Could not resolve your $HOME directory"),
+    }
+}
+
+fn config_dir_path() -> String {
+    match env::home_dir() {
+        Some(home_dir) => format!("{}/{}", home_dir.display(), ".eureka"),
```

Comment:
you may use some more standard directory to save your configuration like: https://specifications.freedesktop.org/basedir-spec/basedir-spec-0.6.html

Reply:
I thought this was pretty standard. `$HOME/.eureka`. If I do `$ cd $HOME && ls -la` I see lots of applications using this format to store their configuration.

The link you provided has lots of information, could you care to explain in a more detailed way where this configuration would be stored?

Reply:
The idea is that there are some "standard" directory where to put application's configuration on unix systems. The best is using `$XDG_CONFIG_HOME/eureka` and fallback to `$HOME/.config/eureka`. I think it's a bit different for Mac OS. And I don't know what's the best way for windows, IIRC there is a directory AppData or something like that.

Reply:
As I said in my previous comment, I'm on Mac and I will have keep Mac users in prio. Porting it to other operating systems is no priority right now, so I will keep the location of where the config is right now because it's very common in macOS.